nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -17,987 +17,569 @@
-            found = true;
-        } else if (bpath.toString().toUpperCase().endsWith(".DX")) {
-            found = true;
+    private List<JCampPage> extractRealPages() {
+        List<JCampPage> realPages = block.getPagesForYSymbol("R");
+        if (!realPages.isEmpty()) {
+            return realPages;
         }
-        return found;
+
+        // Files using XYDATA or Bruker 2D may define "Y" instead of "R" and "I".
+        return block.getPagesForYSymbol("Y");
     }
 
-    @Override
-    public String toString() {
-        return fpath;
+    private List<JCampPage> extractImaginaryPages() {
+        return block.getPagesForYSymbol("I");
     }
 
     @Override
-    public String getVendor() {
-        return "bruker";
+    public void close() {
+        // Nothing to close
     }
 
     @Override
-    public Map getFidFlags() {
-        Map<String, Boolean> flags = new HashMap(5);
-        return flags;
+    public String getFilePath() {
+        return path;
     }
 
     @Override
-    public String getFilePath() {
-        return fpath;
+    public String getPar(String parname) {
+        return block.get(parname).getString();
     }
 
     @Override
-    public DatasetType getPreferredDatasetType() {
-        return preferredDatasetType;
+    public Double getParDouble(String parname) {
+        return block.get(parname).getDouble();
     }
 
     @Override
-    public void setPreferredDatasetType(DatasetType datasetType) {
-        this.preferredDatasetType = datasetType;
+    public Integer getParInt(String parname) {
+        return block.get(parname).getInt();
     }
 
     @Override
     public List<VendorPar> getPars() {
+        Set<String> defined = new HashSet<>();
         List<VendorPar> vendorPars = new ArrayList<>();
-        for (Map.Entry<String, String> par : parMap.entrySet()) {
-            vendorPars.add(new VendorPar(par.getKey(), par.getValue()));
+
+        // get block-level records first
+        for (String key : block.allRecordKeys()) {
+            if (defined.add(key)) {
+                vendorPars.add(new VendorPar(key, block.get(key).getString()));
+            }
+        }
+
+        // then add document-level records if they were not defined by the block
+        for (String key : document.allRecordKeys()) {
+            if (defined.add(key)) {
+                vendorPars.add(new VendorPar(key, document.get(key).getString()));
+            }
         }
+
         return vendorPars;
     }
 
     @Override
-    public String getPar(String parname) {
-        if (parMap == null) {
-            return null;
-        } else {
-            return parMap.get(parname);
-        }
+    public int getNVectors() {
+        //XXX To confirm, looks like it is number of 1D rows in total
+        // so 1 if 1D data, and number of 2D if 2D data.
+        // no more than 2D data are supported in JCamp
+
+        //XXX should we try to get it from page content instead?
```

Comment:
Not sure I completely understood what getNVectors() is supposed to return.

Reply:
getNVectors is the total number of vectors of data in dataset.  1 for 1D, nRows for 2D, nRows*nPlanes for 3D etc.